src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -344,7 +349,14 @@ object ScalafixPlugin extends AutoPlugin {
       if (shellArgs.rules.isEmpty && shellArgs.extra == List("--help")) {
         scalafixHelp
       } else {
-        val scalafixConf = scalafixConfig.in(config).value.map(_.toPath)
+        val scalafixConf =
+          if (scalafixRunExplicitly.value) scalafixConfig.in(config).value
```

Comment:
you can't totally rely on `scalafixRunExplicitly` as it won't be true when scalafix is potentially invoked through another plugin/task, so you should strengthen the check here
```suggestion
          if (scalafixRunExplicitly.value && !scalafixOnCompile.value) scalafixConfig.in(config).value
```

Reply:
I'm afraid that `&& !scalafixOnCompile.value` won't work, because scalafixConfig is intended to be used in case scalafix is explicitly invoked, even if `scalafixOnCompile := true`. 

Please let me describe a use case:

Given that build and scalafix config files are as follows,

```
//build.sbt

scalafixOnCompile := true
scalafixOnCompileConfig := file(".scalafix-on-compile.conf")

semanticdbEnabled := true
semanticdbVersion := scalafixSemanticdb.revision

//.. and other settings
```
```
//.scalafix.conf
rules = [DisableSyntax]
DisableSyntax.keywords = [
  var
  null
]
```

```
//.scalafix-on-compile.conf
rules = [RemoveUnused]
```

On every-compile, a user run DisableSyntax rule for prompt lint without touching source codes.  Occasionally, the user run RemoveUnused rule to remove unused imports or variables by explicit scalafix run.

Reply:
Apologies, that would rather be:
```suggestion
          if (scalafixRunExplicitly.value || !scalafixOnCompile.value) scalafixConfig.in(config).value
```

In your use-case, that would prevent `.scalafix-on-compile.conf` to be used when `scalafix` is called indirectly via a `lintAll` task defined in the build or in a plugin (in which case `scalafixRunExplicitly.value == false`).

Reply:
(I am guessing you flipped `.scalafix.conf` and `.scalafix-on-compile.conf` in your example since `RemoveUnused` is more aggresive than `DisableSyntax` so it probably requires an explicit run - I didn't take that into account in my answer above)

Reply:
> Apologies, that would rather be:
> 
> In your use-case, that would prevent `.scalafix-on-compile.conf` to be used when `scalafix` is called indirectly via a `lintAll` task defined in the build or in a plugin (in which case `scalafixRunExplicitly.value == false`).

Scratch the suggestion, actually it wouldn't prevent that. But there is still this corner case here that we can't really handle...

Reply:
Maybe another way to implement the specific config file for compile, and for compile only, would be to pass `--config ${scalafixOnCompileConfig.in(config.value})` in https://github.com/scalacenter/sbt-scalafix/blob/11d4c3f1bb182c2eb8c7b81ebcb6535f85a3f98e/src/main/scala/scalafix/sbt/ScalafixPlugin.scala#L107. Since the CLI arguments are passed last https://github.com/scalacenter/sbt-scalafix/blob/11d4c3f1bb182c2eb8c7b81ebcb6535f85a3f98e/src/main/scala/scalafix/sbt/ScalafixPlugin.scala#L366, this might work!

Reply:
> when scalafix is potentially invoked through another plugin/task

Indeed. In current implementation, scalafix-invoked-by-another-plugin uses scalafixOnCompileConfig. Since scalafixOnCompileConfig is described as a conf file only for on-compilation hook, it may surprise users.

What do you think to define a task flag like this, and use in this PR?

```scala
 private val scalafixRunOnCompile: Def.Initialize[Task[Boolean]] =
    Def.task {
      executionRoots.value.exists (_.key == compile.key)
    }
```

Reply:
> What do you think to define a task flag like this, and use in this PR?
> 
> ```scala
>  private val scalafixRunOnCompile: Def.Initialize[Task[Boolean]] =
>     Def.task {
>       executionRoots.value.exists (_.key == compile.key)
>     }
> ```

I had thought about it in the implementation of `scalafixOnCompile` - unfortunately you have the same problem: most of the times `compile` is invoked indirectly (via `test` for example) so this is not reliable.

Reply:
Thank you for your throughout support on this.

> Maybe another way to implement the specific config file for compile, and for compile only, would be to pass --config ${scalafixOnCompileConfig.in(config.value})

I'd agree that this way would be the most reliable, so that I've rewrote on this way.